File: python/credit_check.py
def credit_check(str):
    num = [int(i) for i in str]
    doubles_of_every_other = [(( ((num[i]*2) // 10) + ((num[i]*2) % 10) ) if (num[i]*2 > 9) else num[i]*2) if i%2 == 0 else num[i] for i in range(len(num))]
    
    return "The number is valid!" if sum(doubles_of_every_other) % 10 == 0 else "The number is invalid!"

File: js/creditCheck.js
exports.creditCheck = function(str) {
    str = str.split("")
    let sum = 0
    for (let i = 0; i < str.length; i++) {
        if(i%2 == 0) {
            str[i] *= 2
            if (str[i] > 9) {
                str[i] = (str[i] % 10) + parseInt(str[i]/10)
            }
        } else {
            str[i] *= 1
        }
        sum += str[i]
    }
    return (sum % 10 == 0) ? "The number is valid!" : "The number is invalid!"
}


